src/RiotAPILibraries.ts

```diff
@@ -0,0 +1,167 @@
+{
+    valid: boolean,
+    stars: number,
+    library: APILibraryStruct | null,
+    links: string[]
+}
+
+export default class RiotAPILibraries {
+    private bot: Discord.Client;
+    private settings: SharedSettings;
+
+    private lastCall: number;
+
+    private fetchSettings: object;
+
+    constructor(bot: Discord.Client, personalSettings: PersonalSettings, settings: SharedSettings) {
+        console.log("Requested Github extension..");
+        this.bot = bot;
+        this.settings = settings;
+
+        this.bot.on("ready", this.onBot.bind(this));
+        this.bot.on("message", this.onCommand.bind(this));
+
+        this.fetchSettings = {
+            headers: {
+                Accept: "application/json",
+                "Content-Type": "application/json",
+                Authorization: `Basic ${new Buffer(personalSettings.github.username + ":" + personalSettings.github.password, "binary").toString("base64")}`
+            }
+        }
+    }
+
+    onBot() {
+        console.log("Github extension loaded.");
+    }
+
+    async onCommand(message: Discord.Message) {
+
+        const args = message.content.split(" ");
+
+        // if it's just our command, get the list
+        if (args.length < 2) {
+            args.push("list");
+        }
+
+        const [command, language] = args;
+
+        // if !libs, !libraries, etc
+        if (this.settings.githubLibraries.aliases.some(x => x === command)) {
+
+            if (language === "list") {
+                const response = await fetch(this.settings.githubLibraries.baseURL, this.fetchSettings);
+                const data = await response.json() as GithubAPIStruct[];
+
+                let languages = "`" + data.map(x => x.name).join(", ") + "`";
+                let reply = this.settings.githubLibraries.languageList.replace("{languages}", languages);
+                message.channel.send(reply);
+                return;
+            }
+
+            const response = await fetch(this.settings.githubLibraries.baseURL + language);
+            if (response.status != 200) {
+                message.channel.send(this.settings.githubLibraries.githubError + response.status);
+                return;
+            }
+            
+            const libraryList = await response.json();
+            if (!Array.isArray(libraryList) || libraryList.length === 0 || !libraryList[0].sha) {
+                message.channel.send(this.settings.githubLibraries.noLanguage + language);
+                return;
+            }
+
+            let editMessagePromise = message.channel.send(`Found the list of libraries for ${language}, listing ${libraryList.length} libraries, this post will be edited with the result.`);
+
+            let promises: Promise<LibraryDescription>[] = [];
+            for (const library of libraryList) {
+                promises.push(this.describeAPILibrary(library));
+            }
+
+            const libraryDescriptions = (await Promise.all(promises))
```

Comment:
Why not just `await Promise.all(libraryList.map(lib => this.describeAPILibrary(lib)))` instead of doing a manual map operation?

Reply:
Didn't think of it, will do